caf-logging-logback-converters/src/main/java/com/github/cafapi/logging/logback/converters/CustomThrowableProxyConverter.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2019-2024 Open Text.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.cafapi.logging.logback.converters;
+
+import ch.qos.logback.classic.pattern.RootCauseFirstThrowableProxyConverter;
+import ch.qos.logback.classic.spi.IThrowableProxy;
+import ch.qos.logback.classic.spi.StackTraceElementProxy;
+import ch.qos.logback.classic.spi.ThrowableProxyUtil;
+import ch.qos.logback.core.CoreConstants;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class CustomThrowableProxyConverter extends RootCauseFirstThrowableProxyConverter {
```

Comment:
- Classes should be `final` unless you're intentionally leaving open the possibility of using them as base classes.
- They shouldn't be `public` unless they need to be.

```suggestion
final CustomThrowableProxyConverter extends RootCauseFirstThrowableProxyConverter {
```

Reply:
Couldn't accommodate this comment as I am extending `CustomThrowableProxyConverter` for the purpose of tests.

I have addressed all other comments in this PR.